scripts/checkUpdates.sh

```diff
@@ -1,49 +1,87 @@
 #!/bin/bash
 
+has_param() {
+    local term="$1"
+    shift
+    for arg; do
+        if [[ $arg == "$term" ]]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
 check_arch_updates() {
-    official_updates=0
-    aur_updates=0
-	if command -v paru &> /dev/null; then
-		aur_helper="paru"
-	else
-		aur_helper="yay"
-	fi
-
-    if [ "$1" = "-y" ]; then
-        aur_updates=$($aur_helper -Qum 2>/dev/null | wc -l)
-    elif [ "$1" = "-p" ]; then
-        official_updates=$(checkupdates 2>/dev/null | wc -l)
+    if command -v paru &> /dev/null; then
+        aur_helper="paru"
+    else
+        aur_helper="yay"
+    fi
+
+    if has_param "-tooltip" "$@"; then
+        command=" | head -n 50"
+        official_updates=""
+        aur_updates=""
+        sleep 1 # if you use the checkupdates command again too quickly, it will not return anything
```

Comment:
```suggestion
        wait_for_process_to_finish "checkupdates"
```
And we can just call the above function here so we're only waiting until needed.